### ISSUE TITLE
1 enters ball search

### ISSUE THREAD

**Issue by Narcha** [HULKs/hulk#288]
~When our goalie returns from penalized, he enters ball search instead of returning to the goal.~ This might be outdated.

During our game against rUNSWift on 2023-07-06, our keeper left the goal to enter ball search, see https://github.com/HULKs/hulk/issues/288#issuecomment-1623684273

**Comment by MaikRe** [comment]
This can probably be best tested in either another real life game or once #342 is merged so that FOV does in fact apply and robots actually perform look around and searching more accurately.

**Comment by MaikRe** [comment]
At our test game on 21.06.23 this did not seem to be the case anymore, should be explicitly verified though.

**Comment by knoellle** [comment]
Hasn't appeared again in three games.

**Comment by schmidma** [comment]
nope
https://youtube.com/clip/UgkxEMxTDotuOhUgfw2ZC2xLdE18Lik0oE05

**Comment by MaikRe** [comment]
The youtube clip shows 1 entering ball search, but not from penalized as described in the original issue. Should a new issue be opened or should the original issue description be corrected for this situation?

**Comment by Narcha** [comment]
I adjusted the issue title accordingly yesterday.

**Comment by Narcha** [comment]
My current theory as to why this happens:
1. The keeper sees the ball, claims striker.
https://github.com/HULKs/hulk/blob/aba10fc52a572e7733807a7418055f132e5e37ad/crates/control/src/role_assignment.rs#L531-L562

2. The keeper loses the ball, assumes the `Loser` role.
https://github.com/HULKs/hulk/blob/aba10fc52a572e7733807a7418055f132e5e37ad/crates/control/src/role_assignment.rs#L380-L393

3. Then, if no other robot finds the ball before `spl_striker_message_receive_timeout ` elapses, it assumes the `Searcher` role.
4. https://github.com/HULKs/hulk/blob/aba10fc52a572e7733807a7418055f132e5e37ad/crates/control/src/role_assignment.rs#L189-L208

**Comment by schluis** [comment]
Seems to be outdated
